jmx-metrics/src/main/groovy/io/opentelemetry/contrib/jmxmetrics/JmxConnectorHelper.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.contrib.jmxmetrics;
+
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.rmi.NotBoundException;
+import java.rmi.registry.LocateRegistry;
+import java.rmi.registry.Registry;
+import java.util.Map;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.management.remote.JMXConnector;
+import javax.management.remote.JMXConnectorFactory;
+import javax.management.remote.JMXServiceURL;
+import javax.management.remote.rmi.RMIConnector;
+import javax.management.remote.rmi.RMIServer;
+import javax.rmi.ssl.SslRMIClientSocketFactory;
+
+public class JmxConnectorHelper {
+
+  private static final Logger logger = Logger.getLogger(JmxConnectorHelper.class.getName());
+
+  private static RMIServer stub = null;
+  private static final SslRMIClientSocketFactory sslRMIClientSocketFactory =
+      new SslRMIClientSocketFactory();
+
+  private JmxConnectorHelper() {}
+
+  public static boolean validateConfig(final JmxConfig config) throws MalformedURLException {
+
+    if (!config.registrySsl) {
+      // ensure the service URL is valid, as it will be used to connect to the
+      // MBean server when SSL is not enabled on the RMI registry.
+      JMXServiceURL jmxServiceURL = new JMXServiceURL(config.serviceUrl);
+    }
+    return true;
+  }
+
+  public static JMXConnector connect(
```

Comment:
```suggestion
  /**
   * To use SSL, the {@link RMIServer} stub used by the {@link RMIConnector} must be built separately.
   * As a result, we have to unwind the {@link JMXConnectorFactory#connect} method and reimplement pieces.
   */
  public static JMXConnector connect(
```

A helpful comment explaining _why_ we're doing this is useful.

Reply:
great idea, I've added this